src/test/java/org/prebid/server/bidadjustments/FloorAdjustmentsResolverTest.java

```diff
@@ -0,0 +1,242 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.prebid.server.VertxTest;
+import org.prebid.server.bidadjustments.model.BidAdjustmentsRule;
+import org.prebid.server.bidder.model.Price;
+import org.prebid.server.currency.CurrencyConversionService;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.request.ImpMediaType;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mock.Strictness.LENIENT;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.prebid.server.bidadjustments.model.BidAdjustmentType.CPM;
+import static org.prebid.server.bidadjustments.model.BidAdjustmentType.MULTIPLIER;
+import static org.prebid.server.bidadjustments.model.BidAdjustmentType.STATIC;
+import static org.prebid.server.proto.openrtb.ext.request.ImpMediaType.banner;
+import static org.prebid.server.proto.openrtb.ext.request.ImpMediaType.video_outstream;
+
+@ExtendWith(MockitoExtension.class)
+public class FloorAdjustmentsResolverTest extends VertxTest {
+
+    private static final String BIDDER_NAME = "testBidder";
+    private static final String USD = "USD";
+    private static final String EUR = "EUR";
+    private static final String UAH = "UAH";
+
+    @Mock(strictness = LENIENT)
+    private CurrencyConversionService currencyService;
+
+    @Mock(strictness = LENIENT)
+    private BidAdjustmentsRulesResolver bidAdjustmentsRulesResolver;
+
+    private FloorAdjustmentsResolver target;
+
+    @BeforeEach
+    public void before() {
+        target = new FloorAdjustmentsResolver(bidAdjustmentsRulesResolver, currencyService);
+
+        given(currencyService.convertCurrency(any(), any(), any(), any())).willAnswer(invocation -> {
+            final BigDecimal amount = invocation.getArgument(0);
+            final String fromCurrency = invocation.getArgument(2);
+            final String toCurrency = invocation.getArgument(3);
+
+            if (fromCurrency.equals(toCurrency)) {
+                return amount;
+            } else if (toCurrency.equals(USD)) {
+                return amount.divide(BigDecimal.TEN, 4, RoundingMode.HALF_EVEN);
+            } else if (fromCurrency.equals(USD)) {
+                return amount.multiply(BigDecimal.TEN);
+            } else {
+                return amount;
+            }
+        });
```

Comment:
Is it possible to configure `currencyService` for each test with only one action? If so, refactor and configure it individually for each test.